tests/Unit/API/Google/MerchantTest.php

```diff
@@ -0,0 +1,321 @@
+	}
+
+	public function test_claim_website() {
+		$this->assertTrue( $this->merchant->claimwebsite() );
+	}
+
+	public function test_claimwebsite_error() {
+		$this->service->accounts->expects( $this->any() )
+			->method( 'claimwebsite' )
+			->will(
+				$this->throwException(
+					new GoogleException()
+				)
+			);
+
+		$this->expectException( Exception::class );
+		$this->merchant->claimwebsite();
+	}
+
+	public function test_website_already_claimed() {
+		$this->service->accounts->expects( $this->any() )
+			->method( 'claimwebsite' )
+			->will(
+				$this->throwException(
+					new GoogleException( 'claimed', 403 )
+				)
+			);
+
+		$this->expectException( Exception::class );
+        $this->expectExceptionCode( 403 );
+		$this->merchant->claimwebsite();
+	}
+
+	public function test_get_account() {
+		$this->service->accounts->expects( $this->any() )
+			->method( 'get' )
+			->willReturn( $this->createMock( Account::class ) );
+
+		$this->assertInstanceOf( Account::class, $this->merchant->get_account() );
+	}
+
+	public function test_get_account_failure() {
+		$this->service->accounts->expects( $this->any() )
+			->method( 'get' )
+			->will(
+				$this->throwException(
+					new GoogleException( 'error', 400 )
+				)
+			);
+
+		$this->expectException( MerchantApiException::class );
+        $this->expectExceptionCode( 400 );
+		$this->merchant->get_account();
+	}
+
+	public function test_get_accountstatuses() {
+		$this->service->accountstatuses = $this->createMock( Accountstatuses::class );
+		$this->service->accountstatuses->expects( $this->any() )
+			->method( 'get' )
+			->willReturn( $this->createMock( AccountStatus::class ) );
+
+		$this->assertInstanceOf( AccountStatus::class, $this->merchant->get_accountstatus() );
+	}
+
+	public function test_get_accountstatus_failure() {
+		$this->service->accountstatuses = $this->createMock( Accountstatuses::class );
+		$this->service->accountstatuses->expects( $this->any() )
+			->method( 'get' )
+			->will(
+				$this->throwException(
+					new GoogleException( 'error', 400 )
+				)
+			);
+
+		$this->expectException( Exception::class );
+        $this->expectExceptionCode( 400 );
+		$this->merchant->get_accountstatus();
+	}
+
+	public function test_get_productstatuses_batch() {
+		$this->service->productstatuses = $this->createMock( Productstatuses::class );
+
+		$this->service->productstatuses->expects( $this->any() )
+			->method( 'custombatch' )
+			->willReturn( $this->createMock( ProductstatusesCustomBatchResponse::class ) );
+
+		$this->assertInstanceOf(
+			ProductstatusesCustomBatchResponse::class,
+			$this->merchant->get_productstatuses_batch( [ 1, 2, 3 ] )
+		);
+	}
+
+	public function test_update_account() {
+		$account = $this->createMock( Account::class );
+
+		$this->service->accounts->expects( $this->any() )
+			->method( 'update' )
+			->willReturn( $account );
+
+		$this->assertInstanceOf(
+			Account::class,
+			$this->merchant->update_account( $account )
+		);
```

Comment:
Minor: Here, and in a few other places in this class, you could just pass the full object like so:

```suggestion
		$this->assertEquals( $account, $this->merchant->update_account( $account ) );
```
Personally I like this approach because it's stricter.